### PR TITLE
[Docathon][Add Overview Doc No.5-13]补充 Overview

### PR DESCRIPTION
**Pull request by hu-qi**
描述新增了哪些 Overview 的 API 描述、附上中文文档链接

新增设备相关 Overview 及 API 索引

设备设置与属性获取:

- [paddle.device.get_all_custom_device_type](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/get_all_custom_device_type_cn.html)
- [paddle.device.get_all_device_type](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/get_all_device_type_cn.html)
- [paddle.device.get_available_custom_device](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/get_available_custom_device_cn.html)
- [paddle.device.get_available_device](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/get_available_device_cn.html)
- [paddle.device.get_cudnn_version](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/get_cudnn_version_cn.html)
- [paddle.device.set_device](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/set_device_cn.html)
- [paddle.device.get_device](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/get_device_cn.html)

编译环境检测:

- [paddle.device.is_compiled_with_cinn](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/is_compiled_with_cinn_cn.html)
- [paddle.device.is_compiled_with_cuda](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/is_compiled_with_cuda_cn.html)
- [paddle.device.is_compiled_with_custom_deviceEdit](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/is_compiled_with_custom_deviceEdit_cn.html)
- [paddle.device.is_compiled_with_ipu](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/is_compiled_with_ipu_cn.html)
- [paddle.device.is_compiled_with_mlu](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/is_compiled_with_mlu_cn.html)
- [paddle.device.is_compiled_with_npu](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/is_compiled_with_npu_cn.html)
- [paddle.device.is_compiled_with_rocm](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/is_compiled_with_rocm_cn.html)
- [paddle.device.is_compiled_with_xpu](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/is_compiled_with_xpu_cn.html)

设备描述符:
- [paddle.device.IPUPlace](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/IPUPlace_cn.html)
- [paddle.device.MLUPlace](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/MLUPlace_cn.html)
- [paddle.device.XPUPlace](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/XPUPlace_cn.html)

Stream 与 Event 辅助类:
- [paddle.device.Stream](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/Stream_cn.html)
- [paddle.device.Event](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/Event_cn.html)

Stream 与 Event 相关 API:
- [paddle.device.current_stream](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/current_stream_cn.html)
- [paddle.device.set_stream](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/set_stream_cn.html)
- [paddle.device.stream_guard](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/stream_guard_cn.html)
- [paddle.device.synchronize](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/synchronize_cn.html)

CUDA 相关:
- [paddle.device.cuda.Stream](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/cuda/Stream_cn.html)
- [paddle.device.cuda.Event](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/cuda/Event_cn.html)
- [paddle.device.cuda.current_stream](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/cuda/current_stream_cn.html)
- [paddle.device.cuda.device_count](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/cuda/device_count_cn.html)
- [paddle.device.cuda.empty_cache](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/cuda/empty_cache_cn.html)
- [paddle.device.cuda.get_device_capability](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/cuda/get_device_capability_cn.html)
- [paddle.device.cuda.get_device_name](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/cuda/get_device_name_cn.html)
- [paddle.device.cuda.get_device_properties](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/cuda/get_device_properties_cn.html)
- [paddle.device.cuda.max_memory_allocated](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/cuda/max_memory_allocated_cn.html)
- [paddle.device.cuda.max_memory_reserved](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/cuda/max_memory_reserved_cn.html)
- [paddle.device.cuda.memory_allocated](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/cuda/memory_allocated_cn.html)
- [paddle.device.cuda.memory_reserved](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/cuda/memory_reserved_cn.html)
- [paddle.device.cuda.stream_guard](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/cuda/stream_guard_cn.html)
- [paddle.device.cuda.synchronize](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/device/cuda/synchronize_cn.html)

### CODE REVIEW

**Review by Turingg — APPROVED** [review]
LGTM

File: docs/api/paddle/device/Overview_cn.rst [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+编译环境检测
+::::::::::::::::::::
+
+.. csv-table::
+    :header: "API 名称", "API 功能"
+    :widths: 10, 30
+
+    " :ref:`is_compiled_with_cinn <cn_api_paddle_device_is_compiled_with_cinn>` ", "检查 ``whl`` 包是否可以被用来在 CINN 上运行模型"
+    " :ref:`is_compiled_with_cuda <cn_api_paddle_device_is_compiled_with_cuda>` ", "检查 ``whl`` 包是否可以被用来在 GPU 上运行模型"
+    " :ref:`is_compiled_with_custom_deviceEdit <cn_api_paddle_device_is_compiled_with_custom_deviceEdit>` ", "检查 ``whl`` 包是否可以被用来在指定类型的自定义新硬件上运行模型"
```
Comment (sunzhongkai588) [inline docs/api/paddle/device/Overview_cn.rst]:
```suggestion
    " :ref:`is_compiled_with_custom_device <cn_api_paddle_device_is_compiled_with_custom_device>` ", "检查 ``whl`` 包是否可以被用来在指定类型的自定义新硬件上运行模型"
```

**Review by sunzhongkai588 — APPROVED** [review]
LGTM，需要修复一下文档风格来过CI

**Comment by hu-qi** [comment]
> LGTM，需要修复一下文档风格来过CI

奇怪。win 环境下 pre-commit 未检测到，尝试提了 empty commit 。

**Comment by hu-qi** [comment]
> > LGTM，需要修复一下文档风格来过CI
> 
> 奇怪。win 环境下 pre-commit 未检测到，尝试提了 empty commit 。

还在排查中：
image

**Comment by sunzhongkai588** [comment]
> win 环境下 pre-commit 未检测到

安装 pre-commit 了吗？

**Comment by hu-qi** [comment]
> > win 环境下 pre-commit 未检测到
> 
> 安装 pre-commit 了吗？

image

**Pull request merged**